### PR TITLE
New Syntax - Lambdas, Types, and Methods

### PR DESCRIPTION
**Pull request by iamrecursion**
### Pull Request Description
This PR introduces working lambdas, type definitions, top-level bindings and methods for the new syntax. 

### Important Notes
This is done in parallel with the old syntax so nothing is broken during the process. It is also all _staggeringly_ WIP, and will be cleaned up with the final PR that closes the associated task. 

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md), [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

**Comment by iamrecursion** [comment]
Manager tests are majorly flaky on Windows right now. @mwu-tow

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -243,7 +243,9 @@ need to follow these steps:
 4. In the options for that configuration select 'Listen to remote JVM' under
    'Debugger mode:'
 5. Where it provides the command-line arguments for the remote JVM, copy these
-   and add them to `truffleRunOptions` in [`build.sbt`](build.sbt).
+   and add them to `truffleRunOptions` in [`build.sbt`](build.sbt). Remove the
+   portion of these options after `suspend=y`, including the comma. They don't
+   seem to play nicely with GraalVM.
```
Comment (kustosz) [inline CONTRIBUTING.md]: The don't play nicely with anything, those are placeholders to be filled in!

File: build.sbt [hunk 2]
```diff
@@ -342,7 +342,8 @@ val truffleRunOptions = Seq(
   "-Dgraal.TruffleIterativePartialEscape=true",
   "-XX:-UseJVMCIClassLoader",
   "-Dgraal.TruffleBackgroundCompilation=false",
-  "-Dgraalvm.locatorDisabled=true"
+  "-Dgraalvm.locatorDisabled=true",
+  "-agentlib:jdwp=transport=dt_socket,server=n,address=192.168.0.86:12345,suspend=y"
```
Comment (kustosz) [inline build.sbt]: image
Oh, irony :D

File: engine/runtime/src/main/scala/org/enso/compiler/generate/AstToAstExpression.scala [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
-}
+import org.enso.compiler.exception.UnhandledEntity
+import org.enso.interpreter._
 import org.enso.syntax.text.{AST, Debug}
 
 // TODO [AA] Please note that this entire translation is _very_ work-in-progress
 //  and is hence quite ugly right now. It will be cleaned up as work progresses,
 //  but it was thought best to land in increments where possible.
 
+// TODO [MK]
```
Comment (kustosz) [inline engine/runtime/src/main/scala/org/enso/compiler/generate/AstToAstExpression.scala]: both are done

**Pull request merged**